datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/InstitutionalMemoryMetadataMapper.java

```diff
@@ -19,7 +19,15 @@ public InstitutionalMemoryMetadata apply(@Nonnull final com.linkedin.common.Inst
         result.setUrl(input.getUrl().toString());
         result.setDescription(input.getDescription());
         result.setAuthor(input.getCreateStamp().getActor().toString());
+        result.setAuthorUsername(extractAuthorUsername(input.getCreateStamp().getActor().toString()));
         result.setCreated(AuditStampMapper.map(input.getCreateStamp()));
         return result;
     }
+
+    private String extractAuthorUsername(String actor) {
+        if (actor.contains(":")) {
+            return actor.substring(actor.lastIndexOf(":") + 1);
+        }
+        return "";
+    }
```

Comment:
we can get username by creating a OwnerUrn using OwnerUrn.createFromString() and then extracting name from there. Lets do that instead of string parsing.

Reply:
@gabe-lyons : will this stuff work after no-code?

Reply:
OwnerUrn will still exist after no-code, but you're right that we will eventually deprecate and replace it with just the generic Urn class. To use the Urn class, you would create an instance of a generic urn using `Urn.createFromString(rawUrn)`, then access the part of the key you are interested in using `urn.getEntityKey().get(index)`